CMakeLists.txt

```diff
@@ -129,6 +129,11 @@ set(CMAKE_ARCHIVE_OUTPUT_DIRECTORY "${PROJECT_BINARY_DIR}/lib/")
 # Remove rdynamic option
 set(CMAKE_SHARED_LIBRARY_LINK_C_FLAGS )
 
+# Try to determine compiler version if not automatically set by CMake
+if(NOT CMAKE_C_COMPILER_VERSION)
+  EXECUTE_PROCESS(COMMAND ${CMAKE_C_COMPILER} -dumpversion OUTPUT_VARIABLE CMAKE_C_COMPILER_VERSION)
```

Comment:
Two comments:
- Compiler-related scripts are around line 22-33.
- How general is this process execution, how generally is `-dumpversion` supported by compilers?

Reply:
I'll move it up. Regarding the support, after checking, TI doesn't seem to support it, and although Clang accepts the flag it may not output the correct version. I can restrict it to `USING_GCC` if that is okay with you.

Reply:
Sounds OK. But we'll first need confirmation from @mfrey whether it solves the problem. 

(Note: I'm puzzled. How come that cmake cannot identify the gcc version? Gcc support has been there in cmake for ages. If we can query the version via `-dumpversion`, why cannot cmake? Is this some known cmake vs gcc version incompatibility?)

Reply:
I'm guessing the issue is because of `cmake/toolchain_external.cmake:20`.
It says [here] (https://cmake.org/cmake/help/v3.0/module/CMakeForceCompiler.html) that `It also bypasses the check for working compiler and basic compiler information tests`.

Reply:
(Now that we're at it, I'm not quite sure why we're using the external toolchain for the riot-stm32f4 target when we have an stm32f4 toolchain, nevertheless it wouldn't make a difference here, as we're using CMAKE_FORCE_C_COMPILER there as well.)